### ISSUE TITLE
Running under apache reverse proxy

### ISSUE THREAD

**Issue by herraristotle** [jordan-dalby/ByteStash#150]
i can't seem to access the bystash instance via apache reverse proxy,

below is my ssl.conf

        <Location /bytestash>
            # Remove the path from ProxyPass since we're in a Location block
            ProxyPass http://localhost:5000/
            ProxyPassReverse http://localhost:5000/
            RequestHeader set X-Forwarded-Prefix "/bytestash"
            RequestHeader set X-Forwarded-Proto "https"
            RequestHeader set X-Forwarded-Port "443"

            # Add these headers for better proxy handling
            ProxyPreserveHost On
            RequestHeader set X-Real-IP %{REMOTE_ADDR}s
            RequestHeader set X-Forwarded-For %{REMOTE_ADDR}s
        </Location>
        
    below my docker-compose.yaml
    
      bytestash:
    image: ghcr.io/jordan-dalby/bytestash:latest  # Using official image instead of building
    container_name: bytestash
    restart: unless-stopped
    ports:
      - "5000:5000"
    environment:
      # e.g. write /bytestash for a domain such as my.domain/bytestash, leave blank in every other case
      #- BASE_PATH=${BYTESTASH_BASE_PATH}
      # Either provide JWT_SECRET directly or use JWT_SECRET_FILE for Docker secrets
      #- JWT_SECRET_FILE=/run/secrets/jwt
      - JWT_SECRET=${BYTESTASH_JWT_SECRET}
      # how long the token lasts, examples: "2 days", "10h", "7d", "1m", "60s"
      - TOKEN_EXPIRY=${BYTESTASH_TOKEN_EXPIRY}
      # is this bytestash instance open to new accounts being created?
      - ALLOW_NEW_ACCOUNTS=${BYTESTASH_ALLOW_NEW_ACCOUNTS}
      # Should debug mode be enabled? Essentially enables logging, in most cases leave this as false
      - DEBUG=${BYTESTASH_DEBUG}
      # Should we use accounts at all? When enabled, it will be like starting a fresh account so export your snippets, no login required
      - DISABLE_ACCOUNTS=false
      # Should internal accounts be disabled?
      - DISABLE_INTERNAL_ACCOUNTS=false

      # Optional: Enable OIDC for Single Sign On
      #- OIDC_ENABLED=${BYTESTASH_OIDC_ENABLED}
      # Optional: Display name for users signing in with SSO, will default to Single Sign-on
      #- OIDC_DISPLAY_NAME=${BYTESTASH_OIDC_DISPLAY_NAME}
      # Your OIDC issuer url, e.g. https://authentik.mydomain.com/application/o/bytestash/ for authentik
      #- OIDC_ISSUER_URL=${BYTESTASH_OIDC_ISSUER_URL}
      # Your OIDC client ID, you can find it in your app provider
      #- OIDC_CLIENT_ID=${BYTESTASH_OIDC_CLIENT_ID}
      # Your OIDC client secret, again, found in the app provider
      #- OIDC_CLIENT_SECRET=${BYTESTASH_OIDC_CLIENT_SECRET}
      # The OIDC scopes to request, e.g. "openid profile email groups"
      #- OIDC_SCOPES=
    volumes:
      - /opt/ByteStash/data:/data/snippets
# Uncomment to use docker secrets
#    secrets:
#      - jwt

#secrets:
#  jwt:
#    file: ./secrets/jwt.txt

i get the blank page when trying to ccess via my domain eg example.com/bytestash, but a blank page is being displayed

**Comment by jordan-dalby** [comment]
Hi,
Set the BASE_PATH environment variable

**Comment by herraristotle** [comment]
I did, i tried setting and unsetting, still not going through the reverse proxy so i can't access it via domain.com/bytestash, cause i have other service eing proxied through apache.

**Comment by jordan-dalby** [comment]
@herraristotle 
What did you enter for the BASE_PATH?

**Comment by jordan-dalby** [comment]
@herraristotle That's correct. I haven't tried an apache reverse proxy, mine is setup with NPM, could be an apache configuration issue?

**Comment by herraristotle** [comment]
I thought they might be any proxy rewrite I didn't set giving it these issues. Can you share your nginx conf, maybe I can replicate to apache

**Comment by jordan-dalby** [comment]
@herraristotle 
Mine looks like this, I'm using the web UI so I don't really do much in terms of writing the location stuff manually.
```
map $scheme $hsts_header {
    https   "max-age=63072000; preload";
}

server {
  set $forward_scheme http;
  set $server         "192.168.4.186";
  set $port           5000;

  listen 80;
  listen [::]:80;

  listen 443 ssl;
  listen [::]:443 ssl;

  server_name bytestash.redacted;

  # Let's Encrypt SSL
  include conf.d/include/letsencrypt-acme-challenge.conf;
  include conf.d/include/ssl-ciphers.conf;
  ssl_certificate /etc/letsencrypt/live/npm-43/fullchain.pem;
  ssl_certificate_key /etc/letsencrypt/live/npm-43/privkey.pem;

  proxy_set_header Upgrade $http_upgrade;
  proxy_set_header Connection $http_connection;
  proxy_http_version 1.1;

  access_log /data/logs/proxy-host-26_access.log proxy;
  error_log /data/logs/proxy-host-26_error.log warn;

  location / {
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $http_connection;
    proxy_http_version 1.1;
    

    # Proxy!
    include conf.d/include/proxy.conf;
  }

  # Custom
  include /data/nginx/custom/server_proxy[.]conf;
}
```

**Comment by jordan-dalby** [comment]
And proxy.conf
```
## Headers
proxy_set_header Host $host;
proxy_set_header X-Original-URL $scheme://$http_host$request_uri;
proxy_set_header X-Forwarded-Proto $scheme;
proxy_set_header X-Forwarded-Host $http_host;
proxy_set_header X-Forwarded-URI $request_uri;
proxy_set_header X-Forwarded-Ssl on;
proxy_set_header X-Forwarded-For $remote_addr;
proxy_set_header X-Real-IP $remote_addr;

## Basic Proxy Configuration
client_body_buffer_size 128k;
proxy_next_upstream error timeout invalid_header http_500 http_502 http_503; ## Timeout if the real server is dead.
proxy_redirect  http://  $scheme://;
proxy_http_version 1.1;
proxy_cache_bypass $cookie_session;
proxy_no_cache $cookie_session;
proxy_buffers 64 256k;

## Trusted Proxies Configuration
## Please read the following documentation before configuring this:
##     https://www.authelia.com/integration/proxies/nginx/#trusted-proxies
# set_real_ip_from 10.0.0.0/8;
set_real_ip_from 172.19.0.0/16;
# set_real_ip_from 192.168.0.0/16;
# set_real_ip_from fc00::/7;
real_ip_header X-Forwarded-For;
real_ip_recursive on;

## Advanced Proxy Configuration
send_timeout 5m;
proxy_read_timeout 360;
proxy_send_timeout 360;
proxy_connect_timeout 360;
```

**Comment by jordan-dalby** [comment]
@herraristotle any luck?

**Comment by herraristotle** [comment]
Nah. I opted for nginx instead.
